src/app/queries/fetchHelpers.ts

```diff
@@ -43,3 +76,11 @@ export const useAuthorizedFetch = <T>(url: string): QueryFunction<T> => {
   const fetchContext = useFetchContext();
   return () => authorizedFetch(url, fetchContext);
 };
+
+export const useAuthorizedMutate = <T, TData>(
```

Comment:
This still works as a hook because it just returns a function, even though that function is imperative 👍

Reply:
We might want to call it `useAuthorizedPost` though to distinguish it from the react-query mutate stuff